akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -718,6 +753,8 @@ private[stream] abstract class MaterializerSession(val topLevel: StreamLayout.Mo
     new ju.HashMap[InPort, AnyRef] :: Nil
   private var publishersStack: List[ju.Map[OutPort, Publisher[Any]]] =
     new ju.HashMap[OutPort, Publisher[Any]] :: Nil
+  private var matValSrcStack: List[ju.Map[MaterializedValueNode, List[MaterializedValueSource[Any]]]] =
+    new ju.HashMap[MaterializedValueNode, List[MaterializedValueSource[Any]]] :: Nil
```

Comment:
Please note that an empty j.u.HashMap of default capacity can occupy 3.5KB

Source: http://java-performance.info/memory-consumption-of-java-data-types-2/